ros2service/ros2service/api/__init__.py

```diff
@@ -62,3 +65,23 @@ def __call__(self, prefix, parsed_args, **kwargs):
                     if n == service_name:
                         return t
         return service_type_completer()
+
+
+class ServicePrototypeCompleter:
+    """Callable returning a service prototype."""
+
+    def __init__(self, *, service_type_key=None):
+        self.service_type_key = service_type_key
+
+    def __call__(self, prefix, parsed_args, **kwargs):
+        full_service_type = getattr(parsed_args, self.service_type_key)
+        try:
+            parts = full_service_type.split('/')
+            package_name = parts[0]
+            interface_type = parts[1]
+            service_type = parts[-1]
```

Comment:
Similar comment to https://github.com/ros2/ros2cli/pull/301#discussion_r310270381.